# Unir los resultados de la búsqueda de oligosacáridos ramificados y descartar las entradas



In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import os
import pandas as pd

ruta_resultados_oligosacaridos = "/content/drive/MyDrive/TFM/T1/df_resultados_oligosacaridos_ramificados"
ruta_df_harm = "/content/drive/MyDrive/TFM/T1"

df_harm_file = os.path.join(ruta_df_harm,"df_harm_filtro_3.csv")
df_harm = pd.read_csv(df_harm_file)
df_harm.head()

PDB_entry_id    Classification                Organism Uniprot_id Ligand_id  \
0         101M  OXYGEN TRANSPORT  Physeter macrocephalus     P02185       NBN   
1         103M  OXYGEN TRANSPORT  Physeter macrocephalus     P02185       NBN   
2         104M  OXYGEN TRANSPORT  Physeter macrocephalus     P02185       NBN   
3         105M  OXYGEN TRANSPORT  Physeter macrocephalus     P02185       NBN   
4         106M  OXYGEN TRANSPORT  Physeter macrocephalus     P02185       ENC   

                  Ligand_InChi Experimental_method  Resolution  \
0  FSBLVBBRXSCOKU-UHFFFAOYSA-N   X-RAY DIFFRACTION        2.07   
1  FSBLVBBRXSCOKU-UHFFFAOYSA-N   X-RAY DIFFRACTION        2.07   
2  FSBLVBBRXSCOKU-UHFFFAOYSA-N   X-RAY DIFFRACTION        1.71   
3  FSBLVBBRXSCOKU-UHFFFAOYSA-N   X-RAY DIFFRACTION        2.02   
4  JEGVKBYNUPNGJU-UHFFFAOYSA-N   X-RAY DIFFRACTION        1.99   

  Adding_Classification  Affinity  
0                   NaN       NaN  
1                   NaN       NaN  
2                   NaN       NaN  
3                   NaN       NaN  
4                   NaN       NaN

In [ ]:
#@title Unir resultados oligoscaridos ramificados

output_folder = '/content/drive/MyDrive/TFM/T1'
# Crear lista para almacenar los DataFrames
lista_df = []

# Verificar existencia de la carpeta de input
if os.path.exists(ruta_resultados_oligosacaridos):
    # Escanear los archivos en la carpeta
    for i in range(1,10):
      file_path = os.path.join(ruta_resultados_oligosacaridos, f"oligosacaridos_ramificados_{i}.csv")

      # Leer el archivo y agregarlo a la lista
      df = pd.read_csv(file_path, sep=',')
      lista_df.append(df)  # Agregar el DataFrame a la lista

# Unir todos los DataFrames en uno solo
if lista_df:  # Verificar que haya archivos CSV procesados
    df_final = pd.concat(lista_df, ignore_index=True)

    # Definir la ruta del archivo de salida
    output_file = os.path.join(output_folder, "oligosacaridos_ramificados_tot.csv")

    # Guardar el DataFrame en un CSV
    df_final.to_csv(output_file, index=False)
    print(f"Archivo guardado en: {output_file}")
else:
    print("No se encontraron archivos CSV en la carpeta.")

Archivo guardado en: /content/drive/MyDrive/TFM/T1/oligosacaridos_ramificados_tot.csv


In [ ]:
import os
import pandas as pd

output_folder = '/content/drive/MyDrive/TFM/T1'
ruta_resultados_oligosacaridos = '/content/drive/MyDrive/TFM/T1/df_resultados_oligosacaridos_ramificados'


lista_df = []


if os.path.exists(ruta_resultados_oligosacaridos):

    for archivo in os.listdir(ruta_resultados_oligosacaridos):
        if archivo.endswith('.csv'):
            file_path = os.path.join(ruta_resultados_oligosacaridos, archivo)
            df = pd.read_csv(file_path, sep=',')
            lista_df.append(df)


if lista_df:
    df_final = pd.concat(lista_df, ignore_index=True)


    output_file = os.path.join(output_folder, "oligosacaridos_ramificados_tot.csv")


    df_final.to_csv(output_file, index=False)
    print(f"Archivo guardado en: {output_file}")
else:
    print("No se encontraron archivos CSV en la carpeta.")


Archivo guardado en: /content/drive/MyDrive/TFM/T1/oligosacaridos_ramificados_tot.csv


In [ ]:
df_final.shape[0]

4351

Se han encontrado 4351 entradas en que el ligando corresponde a un oligosacarido ramificado.

In [ ]:
df_final.head()

PDB_entry_id Classification                                  Organism  \
0         1A47    GLYCOSIDASE  Thermoanaerobacterium thermosulfurigenes   
1         1A78         LECTIN                         Rhinella arenarum   
2         1AK0   ENDONUCLEASE                      Penicillium citrinum   
3         1AK0   ENDONUCLEASE                      Penicillium citrinum   
4         1AXZ         LECTIN                  Erythrina corallodendron   

  Uniprot_id Ligand_id                 Ligand_InChi Experimental_method  \
0     P26827       ADH  BMZJPVSGERKRHP-ACZMJKKPSA-N   X-RAY DIFFRACTION   
1     P56217       DTT  VHJLVAABSRFDPM-IMJSIDKUSA-N   X-RAY DIFFRACTION   
2     P24289       ADS  RPDDEEQJNPPYRG-CPTYKQRNSA-N   X-RAY DIFFRACTION   
3     P24289       THS  KMPXQZWMYQHTNT-XLPZGREQSA-N   X-RAY DIFFRACTION   
4     P16404       GLA  WQZGKKKJIJFFOK-PHYPRBDBSA-N   X-RAY DIFFRACTION   

   Resolution Adding_Classification  Affinity  es_oligosacarido_ramificado  
0        2.56                   NaN       NaN                         True  
1        2.00                   NaN       NaN                         True  
2        1.80                   NaN       NaN                         True  
3        1.80                   NaN       NaN                         True  
4        1.95                   NaN       NaN                         True

In [ ]:
#@title Crear lista entrada y descartar

lista_oligosacaridos = df_final["PDB_entry_id"].tolist()
print(f"Elementos en df antes de aplicar el filtro: {df_harm.shape[0]}")

df_harm_filtro_4 = df_harm[~df_harm["PDB_entry_id"].isin(lista_oligosacaridos)]
print(f"Elementos en df después de aplicar el filtro: {df_harm_filtro_4.shape[0]}")
print(f"Se han descartado {len(lista_oligosacaridos)} entradas")

Elementos en df antes de aplicar el filtro: 95919
Elementos en df después de aplicar el filtro: 91568
Se han descartado 4351 entradas


In [ ]:
#@title Guardar el archivo filtrado (Sin oligosacáridos ramificados)

output_folder = '/content/drive/MyDrive/TFM/T1'
output_file = os.path.join(output_folder, "df_harm_filtro_4.csv")
df_harm_filtro_4.to_csv(output_file, index=False)
print(f"Archivo guardado en: {output_file}")


Archivo guardado en: /content/drive/MyDrive/TFM/T1/df_harm_filtro_4.csv
